Add start, end time and recurrences for schedules

alexVasylenko opened the pull request:
Hi! I want to add a possibility to set the time range when this schedule should work. Also a user can set how many times this schedule should be executed instead of time range.
![initial](https://user-images.githubusercontent.com/59963571/167571596-0f94559c-896a-4e1f-8aca-fa619a6d650a.png)
![timeRange](https://user-images.githubusercontent.com/59963571/167571644-c4da5dbc-0d0c-40db-9bb4-91272835327c.png)
![withDate](https://user-images.githubusercontent.com/59963571/167571672-8fa44f8d-9bbc-4eae-a5a8-e0ab2e308a47.png)
![occur](https://user-images.githubusercontent.com/59963571/167571697-8dffe551-5e2b-413c-b7f8-b1d25b541efc.png)
![occur-result](https://user-images.githubusercontent.com/59963571/167571716-cbf6fc4f-1126-4fb5-9f97-98ead289c2fb.png)

czgu commented on querybook/webapp/components/DataDocSchedule/DataDocScheduleForm.tsx:
```diff
@@ -226,6 +286,23 @@ export const DataDocScheduleForm: React.FunctionComponent<IDataDocScheduleFormPr
                             <Form>
                                 <DisabledSection disabled={!isEditable}>
+                                    <ExecutionRangeField
+                                        values={formatExecutionRangeValue(
```
seems like formatExecutionRangeValue achieves nothing since the `ExecutionRangeField` expects all 3 fields (whether null or not) anyway

czgu commented on querybook/webapp/components/ExecutionRangeField/ExecutionRangeField.tsx:
```diff
@@ -0,0 +1,291 @@
+
+function formatDefaultDateValue(value) {
+    const defaultValue = value ?? moment().format('X');
+    return moment(defaultValue, 'X');
```
`return value ? moment(value, 'X') : moment()`

czgu commented on querybook/webapp/components/ExecutionRangeField/ExecutionRangeField.tsx:
```diff
@@ -0,0 +1,291 @@
+${formatTimeLabel(timeRange.endTime)}`;
+}
+
+function formatDefaultDateValue(value) {
```
getValueOrNow

czgu commented on querybook/webapp/components/ExecutionRangeField/ExecutionRangeField.tsx:
```diff
@@ -0,0 +1,291 @@
+    values: IDate | IOccurence;
+}
+
+export const ExecutionRangeField: React.FunctionComponent<IExecutionRangeFieldProps> = ({
```
can you rename this to ScheduleRangeField since its not related to execution here?

czgu commented on querybook/webapp/components/ExecutionRangeField/ExecutionRangeField.tsx:
```diff
@@ -0,0 +1,291 @@
+    });
+
+    const onSaveCallback = useCallback(
+        (event) => {
```
you don't need an event here (use standard Button component)

czgu commented on querybook/webapp/components/ExecutionRangeField/ExecutionRangeField.tsx:
```diff
@@ -0,0 +1,291 @@
+
+            updateValues(values);
+
+            dispatch({
```
if no fields is mutated, there is no need to save?

czgu commented on querybook/webapp/components/ExecutionRangeField/ExecutionRangeField.tsx:
```diff
@@ -0,0 +1,291 @@
+    }, [startTime, endTime, occurences]);
+
+    const onChangeDatePicker = useCallback(
+        (value, key) => {
```
add typing for callbacks

czgu commented on querybook/webapp/components/ExecutionRangeField/ExecutionRangeField.tsx:
```diff
@@ -0,0 +1,291 @@
+        [state.dateRange]
+    );
+
+    const onInputChange = useCallback(
```
handleOccurencesChange

czgu commented on querybook/webapp/components/ExecutionRangeField/ExecutionRangeField.tsx:
```diff
@@ -0,0 +1,291 @@
+        []
+    );
+
+    const onSelect = useCallback(
```
handleTabSelect

czgu commented on querybook/webapp/components/ExecutionRangeField/ExecutionRangeField.tsx:
```diff
@@ -0,0 +1,291 @@
+                                onChange={onInputChange}
+                            />
+                        )}
+                        <StyledBtn onClick={onSaveCallback}>Save</StyledBtn>
```
since this is getting used in a form which has "Save", can you make this "Confirm" to make it more clear?

czgu commented on querybook/webapp/components/ExecutionRangeField/ExecutionRangeField.tsx:
```diff
@@ -0,0 +1,291 @@
+                        )}
+                        <StyledBtn onClick={onSaveCallback}>Save</StyledBtn>
+                        <StyledButton onClick={onClearCallback}>
+                            Clear
```
the clear button here doesn't really "clear" the form values, can we use "cancel" instead?

czgu commented:
Hey Alex, I have some questions about functionality:

- how can users view how many occurrences there are left?
- How would occurrences be reset?
- If I did set an execution range, how would I remove it in future?

alexVasylenko commented:
> Hey Alex, I have some questions about functionality:
> 
> * how can users view how many occurrences there are left?
> * How would occurrences be reset?
> * If I did set an execution range, how would I remove it in future?

Hi! For the first question, I have added an amount of how many times this DataDoc was executed.
Second, a user can set occurrences now and then can set a date range, and after that,
![test empty - Querybook 2022-07-05 07-51-08](https://user-images.githubusercontent.com/59963571/177326786-915872ae-e801-439f-97b9-e5c6f1a98640.png)
![test empty - Querybook 2022-07-05 07-51-46](https://user-images.githubusercontent.com/59963571/177326823-5bfa0846-bff6-4a4b-b043-57e7aa7495c6.png)
 it would be a data range.
Finally, I added a button that can remove the current execution range. (See attached files)
